sympy/matrices/tests/test_matrices.py

```diff
@@ -196,20 +196,20 @@ def test_power():
     assert Matrix([[1, 2], [3, 4]])**Integer(2) == Matrix([[7, 10], [15, 22]])
 
     A = Matrix([[33, 24], [48, 57]])
-    assert (A**(S(1)/2))[:] == [5, 2, 4, 7]
+    assert ( (A**(S(1)/2)).doit() )[:] == [5, 2, 4, 7]
     A = Matrix([[0, 4], [-1, 5]])
-    assert (A**(S(1)/2))**2 == A
+    assert ((A**(S(1)/2))**2).doit() == A
```

Comment:
Should I implement a workaround where , if a MatPow object is yet again raised to another exponent ; it's checked if clubbing together exponents yields an integer ? If so , then evaluate.

Example:
`(A ** (S(1)/2))**3` --> `MatPow(A,S(1)/2)` is first created.Since clubbing together exponents results in `3/2` , final result returned is `MatPow(A,S(3)/2)`.

`(A ** (S(1)/3))**6` -->  `MatPow(A,S(1)/3)` is first created.Since clubbing together exponents results in `2` , final result returned is `A ** 2` (evaluated form).